### PR TITLE
Integrate LLVM at llvm/llvm-project@b2505ca2ece3

### PR DESCRIPTION
**Pull request by smit-hinsu**

### CODE REVIEW

**Review by burmako — APPROVED** [review]

**Comment by smit-hinsu** [comment]
Please wait for the build to finish before merge as I haven't built locally. Wanted to start the build on github as it will take some time.

**Pull request merged**
